python/lsst/ts/observatory/control/maintel/mtcs.py

```diff
@@ -841,9 +842,40 @@ async def close_m1_cover(self) -> None:
                 f"{MTMount.DeployableMotionState.DEPLOYED!r}"
             )
 
-    async def home_dome(self) -> None:
-        # TODO: Implement (DM-21336).
-        raise NotImplementedError("# TODO: Implement (DM-21336).")
+    async def home_dome(self, physical_az: float = 0.0) -> None:
+        """Utility method to home dome.
+
+        Parameters
+        ----------
+        physical_az : `float`
+            Azimuth angle of the dome as read by markings (in deg).
+
+        """
+        self.log.info("Homing dome")
+        reported_az = await self.rem.mtdome.tel_azimuth.aget(timeout=self.fast_timeout)
+
+        offset = physical_az - reported_az.positionActual
+        self.log.debug(f"Dome azimuth offset: {offset} degrees")
+        target_az = self.home_dome_az - offset
+        await self.slew_dome_to(target_az)
+
+        await self.rem.mtdome.cmd_stop.set_start(
+            engageBrakes=True,
+            subSystemIds=MTDome.SubSystemId.AMCS,
+            timeout=self.long_long_timeout,
+        )
+        motion_state = await self.rem.mtdome.evt_azMotion.aget(
+            timeout=self.fast_timeout
+        )
+        while motion_state.state != MTDome.MotionState.STOPPED_BRAKED:
+            motion_state = await self.rem.mtdome.evt_azMotion.next(
+                flush=False, timeout=self.fast_timeout
```

Comment:
I think `fast_timeout` is too short here.. you probably want something like `long_long_timeout`